plugins/BEdita/Core/src/Model/Behavior/UniqueNameBehavior.php

```diff
@@ -116,9 +133,27 @@ public function uniqueNameExists($uname, $id = null)
      * @param \Cake\Event\Event $event The event dispatched
      * @param \Cake\Datasource\EntityInterface $entity The entity to save
      * @return void
+     * @codeCoverageIgnore
      */
     public function beforeSave(Event $event, EntityInterface $entity)
     {
         $this->uniqueName($entity);
     }
+
+    /**
+     * Setup initial unique name when a new Entity is being created with empty `uname` field
+     *
+     * @param \Cake\Event\Event $event The event dispatched
+     * @param ArrayObject $data The input data to save
+     * @param ArrayObject $options Operation options (unused)
+     * @return void
+     */
+    public function beforeMarshal(Event $event, ArrayObject $data, ArrayObject $options)
+    {
+        if (empty($data['uname']) && empty($data['id'])) {
+            $field = $this->config('sourceField');
+            $fieldValue = !empty($data[$field]) ? $data[$field] : (!empty($data['type']) ? $data['type'] : '');
```

Comment:
I know this might sound kind of pedantic, but I would prefer splitting this into multiple lines, like:
```php
$fieldValue = '';
if (!empty($data[$field])) {
    $fieldValue = $data[$field];
} elseif (!empty($data['type'])) {
    $fieldValue = $data['type'];
}
```

Apart from readability, with is subjective, doing this makes debugging slightly easier — and line coverage would become closer to branch coverage, which is an important metric that's not supported by XDebug, so it's better to avoid complex single-line conditions.